odl/discr/discr_ops.py

```diff
@@ -130,110 +144,212 @@ def finite_diff(f, out=None, axis=0, dx=1.0, edge_order=2,
 
     method, method_in = str(method).lower(), method
     if method not in ('central', 'forward', 'backward'):
-        raise ValueError("method '{}' has to be central, forward or backward"
+        raise ValueError('method {} is not central, forward or backward'
                          ''.format(method_in))
 
+    if isinstance(padding, (int, float)) or padding in (None, 'edge'):
+        if isinstance(padding, int):
```

Comment:
The problem with `padding = float(padding)` is that if one uses `True` or `False`, you will get padding with 1 or 0, respectively. So one needs to check that too. Maybe its better to split the padding method and the padding value (if required), also with respect to other padding methods to be implemented in the future.